pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/io/PinotOutputFormat.java

```diff
@@ -0,0 +1,255 @@
+    }
+
+    public static void getHllSize(Job job, int size) {
+        job.getConfiguration().setInt(PinotOutputFormat.HLL_SIZE, size);
+    }
+
+    public static int getHllSize(JobContext context) {
+        return context.getConfiguration().getInt(PinotOutputFormat.HLL_SIZE, 9);
+    }
+
+    public static void setHllColumns(Job job, String columns) {
+        job.getConfiguration().set(PinotOutputFormat.HLL_COLUMNS, columns);
+    }
+
+    public static String getHllColumns(JobContext context) {
+        return context.getConfiguration().get(PinotOutputFormat.HLL_COLUMNS);
+    }
+
+    public static void getHllSuffix(Job job, String suffix) {
+        job.getConfiguration().set(PinotOutputFormat.HLL_SUFFIX, suffix);
+    }
+
+    public static String getHllSuffix(JobContext context) {
+        return context.getConfiguration().get(PinotOutputFormat.HLL_SUFFIX, HllConstants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX);
+    }
+
+    public static void setDataWriteSupportClass(Job job, Class<? extends PinotRecordSerialization> pinotSerialization) {
+        job.getConfiguration().set(PinotOutputFormat.PINOT_RECORD_SERIALIZATION_CLASS, pinotSerialization.getName());
+    }
+
+    public static Class<?> getDataWriteSupportClass(JobContext context) {
+        String className = context.getConfiguration().get(PinotOutputFormat.PINOT_RECORD_SERIALIZATION_CLASS);
+        if (className == null) {
+            throw new RuntimeException("pinot data write support class not set");
+        }
+        try {
+            return context.getConfiguration().getClassByName(className);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public RecordWriter<K, V> getRecordWriter(TaskAttemptContext context) throws IOException, InterruptedException {
+        configure(context.getConfiguration());
+        final PinotRecordSerialization dataWriteSupport = getDataWriteSupport(context);
+        initSegmentConfig(context);
+        Path workDir = getDefaultWorkFile(context, "");
+        return new PinotRecordWriter<>(_segmentConfig, context, workDir, dataWriteSupport);
+    }
+
+    /**
+     * The {@link #configure(Configuration)} method called before initialize the  {@link
+     * RecordWriter} Any implementation of {@link PinotOutputFormat} can use it to set additional
+     * configuration properties.
+     */
+    public void configure(Configuration conf) {
+
+    }
+
+    private PinotRecordSerialization getDataWriteSupport(TaskAttemptContext context) {
+        try {
+            return (PinotRecordSerialization) PinotOutputFormat.getDataWriteSupportClass(context).newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Error initialize data write support class", e);
+        }
+    }
+
+    private void initSegmentConfig(JobContext context) throws IOException {
+        _segmentConfig.setFormat(FileFormat.JSON);
```

Comment:
Unfortunately, have to store the intermediate file as a json format for a couple of reason
1. GenericRow is not a recognized FileFormat
2. Stream/bytes are not supported as a default contract for Pinot